src/vm.rs

```diff
@@ -457,7 +457,7 @@ pub fn step_vm<'gc>(
                 } => {
                     let left = registers.stack_frame[left.0 as usize];
                     let right = registers.stack_frame[right.0 as usize];
-                    if (left < right) == skip_if {
+                    if (left < right) != skip_if {
```

Comment:
So I fixed the issue with the less than operator.  But I don't like how I fixed it.  Because I don't understand how it can be possible that this is different from the equality operator.  You said out of thread that even though `== skip_if`is different from Lua, it ought to be fine as long as its consistent.  Which makes me think that there's some bug in the compiler generating something backwards, and I undid the backward bug, rather than fixing the issue the right way.  But I've spent a few hours looking at it, and fiddling with it, but I can't find it.

Reply:
The compiler is probably reversing the arguments, you wouldn't see a bug with equals because it's symmetric

Reply:
Never mind, found it, see above review comment